addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -530,11 +530,16 @@ function factory(dependencies) {
             // 1. move messages from inbox to history
             // AKU TODO: flag other caches to invalidate
             // task-2171873
-            inboxMailbox.messages.map(message => {
-                message.update({
-                    isNeedaction: false,
-                    isHistory: true,
-                });
+            message_ids.map(message_id => {
+                const message = inboxMailbox.messages.find(
+                    inboxMessage => inboxMessage.id === message_id
+                );
```

Comment:
As discussed, we can get rid of filtering only inbox messages.

You can do `mail.message` `insert` to find them all.

Reply:
The test `QUnit.test('all messages in "Inbox" in "History" after marked all as read'` don't pass anymore when I do the change.
The reason is that in its mockRPC there is : 
```js
if (args.method === 'mark_all_as_read') {
                const messageIDs = [];
                for (let i = 0; i < messagesData.length; i++) {
                    this.data['mail.message'].records[i].history_partner_ids = [3];
                    this.data['mail.message'].records[i].needaction_partner_ids = [];
                    this.data['mail.message'].records[i].needaction = false;
                    messageIDs.push(messageOffset + i);
                }
                const notificationData = {
                    type: 'mark_as_read',
                    message_ids: messageIDs,
                };
                const notification = [[false, 'res.partner', 3], notificationData];
                self.widget.call('bus_service', 'trigger', 'notification', [notification]);
                markAllReadDef.resolve();
                return 3;
            }
```

That means that when showing 30 messages in the inbox, it returns a "mark as read" notifications for 40 messages and as we do not filter anymore in notification handling, the whole messages (even the 10 not yet loaded & shown) are marked as read.

So, do we have to change that test or the way we handle the notification ?

Reply:
I'm changing this test in my branch to work with data and mock server, so its result should become reliable in all cases. Not sure if it's worth doing anything else in the meantime.

Reply:
But for the handler, we might want to ignore messages that are not known, so using filter on .all()

Reply:
In fact the problem is indeed that we use "insert. So we could do with a find instead ?

Reply:
> But for the handler, we might want to ignore messages that are not known, so using filter on .all()

Or insert these messages and flag them as read (= `isHistory`). We are aware of existence of these messages, so we could store them. They are not linked to thread cache so they won't be displayed in a weird way.

Reply:
Yeah I know but then the test should be adapted as it should be 40 messages directly in history then